paper/ml.tex

```diff
@@ -2,7 +2,68 @@ \section{AI and Machine Learning Contributions}\label{ml-sec}
+  5/5/90 & 99.6 \\\hline
+  1/1/98 & 99.3 \\\hline
+  0.5/0.5/99 & 98.9 \\\hline
+  0.1/0.1/99.8 & 92.2 \\
+  \hline
+\end{tabular}
+\caption{Prediction accuracy as function of size of training set}
+\end{table}
+
+\noindent Since these training datasets were sensibly smaller than the subset of explicit implications, and were not carefully chosen from the poset extremes but taken randomly, we can conclude that even if the CNN was learning transitivity, probably this alone is not enough to explain the high accuracy achieved by the CNN model.
+
+\smallskip
+
+Since sometimes machine learning is announced as a form of data compression, let us now comment on the level of data compression achieved by our CNN model. In the table below we compare the sizes of three different encodings of the implication graph: a) As the simplest approach, we can encode the full implication graph in one file as labelled pairs of equations of the form ($p$, $q$, yes/no). b) On the other extreme, we can encode it as a bit table not containing neither the explicit equation expressions nor their numbers, but just a 1/0 label for each point with coordinates ($p$,$q$), together with a table mapping each number to its corresponding equation expression and a small script to recover the file in a). c) Finally, we can encode it in the complete files of the CNN model produced by TensorFlow 2.9.0. In addition, we can either consider these files in their raw form, or we can highly (but losslessly) compress them to achieve a rough comparison of their actual information content; accordingly, in the table below we also include the sizes of the encoding models when compressed with 7-zip LZMA2 ultra compression with a 1536MB dictionary size and a 273 word size.
+
+\begin{table}[h]
+\centering
+\begin{tabular}{|c|c|c|}
+  \hline
+  % after \\: \hline or \cline{col1-col2} \cline{col3-col4} ...
+  \textbf{Encoding model} & \textbf{Uncompressed size} & \textbf{Compressed size} \\\hline\hline
+  Labelled pairs of equations & 1.5GB & 9MB \\\hline
+  Bit table & 42MB & 40KB \\\hline
+  CNN (99.7\% accuracy) & 1.34MB & 700KB \\
+  \hline
+\end{tabular}
+\caption{Size of different encodings for the implication graph}
+\end{table}
+
+\smallskip
+
+\noindent As we see, the information in the CNN model is more than 13 times less than in the labelled pairs model, while it is more than 18 times that of the bit table model. We also note that the CNN model in its raw form is already quite incompressible, and more than 30 times smaller than the raw bit table.
+
+\smallskip
+
+Lastly, take also into account that our CNN model does not only encode the implication graph up to order 4 (with a 0.03\% of noise) but a priori may also be able to predict it for higher orders with a significant accuracy, and could thus be used to guide and speed up the determination of the implication graph up to order 5, by letting ATPs focus first on the CNN’s predicted status of the studied implication.
```

Comment:
"also take into account" is better, although for some reason "take into account" is an unusual imperative in a paper.

Could also break the sentence at "and could thus", has lots of subparts now.